raiden/network/transport/matrix.py

```diff
@@ -584,53 +599,57 @@ def _send_immediate(self, receiver_address, data):
 
     def _get_room_for_address(
         self,
-        receiver_address: Address,
+        address: Address,
         allow_missing_peers=False,
     ) -> Optional[Room]:
         if not self._running:
             return
-        room_id = self._address_to_roomid.get(receiver_address)
+        address_hex = to_normalized_address(address)
+        assert address and address in self._address_to_userids,\
+            f'address not health checked: me: {self._client.user_id}, peer: {address_hex}'
+        room_id = self._get_room_id_for_address(address)
         if room_id:
-            room = self._client.rooms.get(room_id)
-            if room:
-                return room
-            else:
-                # Room is gone - remove from cache
-                self._address_to_roomid.pop(receiver_address)
+            return self._client.rooms[room_id]
 
         # The addresses are being sorted to ensure the same channel is used for both directions
         # of communication.
         # e.g.: raiden_ropsten_0xaaaa_0xbbbb
         address_pair = sorted([
             to_normalized_address(address)
-            for address in [receiver_address, self._raiden_service.address]
+            for address in [address, self._raiden_service.address]
         ])
         room_name = self._make_room_alias(*address_pair)
 
-        room_candidates = self._client.search_room_directory(room_name)
-        if room_candidates:
-            room = room_candidates[0]
-            if room.room_id not in self._client.rooms:
-                room = self._client.join_room(room.room_id)
-        else:
-            # no room with expected name => create one and invite peer
-            address = to_normalized_address(receiver_address)
-            candidates = self._client.search_user_directory(address)
-
-            # filter candidates
-            peers = [user for user in candidates if self._validate_userid_signature(user)]
-            if not peers and not allow_missing_peers:
-                self.log.error('No valid peer found', peer_address=address)
-                return
+        # no room with expected name => create one and invite peer
```

Comment:
Is this comment still accurate?

Reply:
Yes, because although room name isn't enforced, it's still the first try.

Reply:
Ok